DEVELOPMENT.md

```diff
@@ -1,85 +1,263 @@
+then we suggest going over the [software architecture](docs/software_arch.md) for an overview of the components and how they function.
+
+## The Development Loop
+
+We have configured [hatch](https://github.com/pypa/hatch) commands to support a standard development loop. You can run the following
+from any directory of this repository:
+
+* `hatch build` - To build the installable Python wheel and sdist packages into the `dist/` directory.
+* `hatch run test` - To run the PyTest unit tests found in the `test/unit` directory. See [Testing](#testing).
+* `hatch run all:test` - To run the PyTest unit tests against all available supported versions of Python.
+* `hatch run lint` - To check that the package's formatting adheres to our standards.
+* `hatch run fmt` - To automatically reformat all code to adhere to our formatting standards.
+* `hatch shell` - Enter a shell environment that will have Python set up to import your development version of this package.
+* `hatch env prune` - Delete all of your isolated workspace [environments](https://hatch.pypa.io/1.12/environment/) 
+   for this package.
+
+Note: Hatch uses [environments](https://hatch.pypa.io/1.12/environment/) to isolate the Python development workspace
+for this package from your system or virtual environment Python. If your build/test run is not making sense, then
+sometimes pruning (`hatch env prune`) all of these environments for the package can fix the issue.
+
+### One time plugin-in Environment Setup
+These instructions make the following assumptions:
+
+* You have a [git clone of this repository](https://docs.github.com/en/repositories/creating-and-managing-repositories/cloning-a-repository#cloning-a-repository)
+* You have `pip` available in your terminal
+* And you have run:
+`pip install --upgrade -r requirements-development.txt`
+
+Before you begin plug-in development, you'll need complete a couple one-time steps:
+
+1. Install plug-in dependencies to a development location:
+   - For Blender 3.6-4.0 (uses python 3.10):
+      - Windows: `pip install --python-version 3.10 --only-binary=:all: "deadline[gui]" blender-qt-stylesheet -t %USERPROFILE%\DeadlineCloud\Blender\python\modules`
+      - Linux/macOS: `pip install --python-version 3.10 --only-binary=:all: "deadline[gui]" blender-qt-stylesheet -t ~/DeadlineCloud/Blender/python/modules`
+   - For Blender 4.1-4.2 (uses python 3.11):
+      - Windows: `pip install --python-version 3.11 --only-binary=:all: "deadline[gui]" blender-qt-stylesheet -t %USERPROFILE%\DeadlineCloud\Blender\python\modules`
+      - Linux/macOS: `pip install --python-version 3.11 --only-binary=:all: "deadline[gui]" blender-qt-stylesheet -t ~/DeadlineCloud/Blender/python/modules`
+
+2. Add a script directory in Blender by "Edit" > "Preferences" > "File Paths" > "Script Directories"
+   * Windows: `%USERPROFILE%\DeadlineCloud\Blender\python`
+   * Linux/macOS: `~/DeadlineCloud/Blender/python`
```

Comment:
Since this is specifically for developement and not a generic install, does this not have to be the location of the git repo location? The instructions don't tell you to run hatch build -> and pip install the whl. But they also don't point to the addon source directory. Looks like this was deleted from the previous instructions:

```
1. Install addon in Blender
    - run `hatch build` in your local git repository
    - `cp -r src/deadline/blender_submitter/addons/ ~/DeadlineCloudSubmitter/Submitters/Blender/python/addons`
```

Also, if they make changes, will they have to run commands to get their changes to show? We should include that in the workflow instructions.

We may need a separate section for manual installing for using, not for developing. ie. pip install from pypi, don't need clone of github, etc.